src/spag/main.rs

```diff
@@ -22,21 +27,36 @@ use super::yaml_util;
 
 
 pub fn main() {
-    let args: Args = args::parse_args();
-    if args.cmd_request {
-        spag_request(&args);
-    } else if args.cmd_history {
-        spag_history(&args);
-    } else if args.cmd_env {
-        spag_env(&args);
-    } else if args.cmd_get || args.cmd_post || args.cmd_put || args.cmd_patch || args.cmd_delete {
-        spag_method(&args);
+    let argv: Vec<String> = std::env::args().collect();
+    let main_argv: Vec<String> = std::env::args().take(2).collect();
+    let args: MainArgs = args::parse_main_args(&main_argv);
+    match args.arg_command.as_str() {
+        "env" => {
+            spag_env(&args::parse_env_args(&argv))
+        },
+        "r" | "request" => {
+            spag_request(&args::parse_request_args(&argv))
+        },
+        "history" => {
+            spag_history(&args::parse_history_args(&argv))
+        },
+        "get" | "post" | "put" | "patch" | "delete" => {
+            spag_method(&args::parse_method_args(&argv))
+        },
+        command if command.is_empty() => {
+            printerrln!("Received no command or options");
```

Comment:
Yeah... I can't do this without changing all the usage strings to have `spag env <command>` instead of `spag env set`, `spag env unset`, etc. Or by changing the `parse_args!` macro to return something different, which I don't want to do.